src/code/audio_synthesis.c

```diff
@@ -1184,27 +1189,33 @@ Acmd* AudioSynth_ProcessEnvelope(Acmd* cmd, NoteSubEu* noteSubEu, NoteSynthesisS
     return cmd;
 }
 
-Acmd* AudioSynth_LoadWaveSamples(Acmd* cmd, NoteSubEu* noteSubEu, NoteSynthesisState* synthState, s32 nSamplesToLoad) {
-    s32 temp_v0;
-    s32 unk6 = noteSubEu->unk_06;
+Acmd* AudioSynth_LoadWaveSamples(Acmd* cmd, NoteSubEu* noteSubEu, NoteSynthesisState* synthState,
+                                 s32 numSamplesToLoad) {
+    s32 numSampleSlotsAvail;
+    s32 harmonicIndexCurAndPrev = noteSubEu->harmonicIndexCurAndPrev;
     s32 samplePosInt = synthState->samplePosInt;
-    s32 repeats;
+    s32 numDuplicates;
 
     if (noteSubEu->bitField1.bookOffset != 0) {
-        AudioSynth_LoadBuffer(cmd++, DMEM_UNCOMPRESSED_NOTE, ALIGN16(nSamplesToLoad * 2), gWaveSamples[8]);
-        gWaveSamples[8] += nSamplesToLoad * 2;
+        AudioSynth_LoadBuffer(cmd++, DMEM_UNCOMPRESSED_NOTE, ALIGN16(numSamplesToLoad * sizeof(s16)), gWaveSamples[8]);
+        gWaveSamples[8] += numSamplesToLoad * sizeof(s16);
         return cmd;
     } else {
-        aLoadBuffer(cmd++, noteSubEu->sound.samples, DMEM_UNCOMPRESSED_NOTE, 0x80);
-        if (unk6 != 0) {
-            samplePosInt = (samplePosInt * D_801304C0[unk6 >> 2]) / D_801304C0[unk6 & 3];
+        // move the synthetic wave from ram to the rsp
+        aLoadBuffer(cmd++, noteSubEu->sound.waveSampleAddr, DMEM_UNCOMPRESSED_NOTE, 64 * sizeof(s16));
+        if (harmonicIndexCurAndPrev != 0) {
+            // samplePosInt scaled by (current-harmonic / prev-harmonic)
+            samplePosInt = samplePosInt * sNumSamplesPerWavePeriod[harmonicIndexCurAndPrev >> 2] /
+                           sNumSamplesPerWavePeriod[harmonicIndexCurAndPrev & 3];
```

Comment:
:thinking: not sure about the comment, I would guess it means `by (current-harmonic-order / prev-harmonic-order)`, but that would be wrong since the higher the harmonic order the lower the period

Reply:
I've rephrased it to emphasize more about mapping the offset from one harmonic to another.

As an example, let's say the peak of a wave is 1/4 of the way through a single period. For harmonic index = 0, that means this peak is 64 / 4 = 16 samples into the wave. However, for harmonic index = 0, the period is over 32 samples, so 32 / 4 = 8.

So `samplePosInt` is the offset in a single period as to where to start processing the wave. But you want to be in the same spot for different harmonics, so that's what this mapping does. Does that make sense? It's clear to me now what this is doing, but let me know if you find a better way to phrase this with a short amount of words